flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestMetadataTableReadableMetrics.java

```diff
@@ -291,7 +291,7 @@ public void testSelectNestedValues() throws Exception {
   public void testNestedValues() throws Exception {
     createNestedTable();
 
-    Row leafDoubleCol = Row.of(53L, 3L, 1L, 1L, 0.0D, 0.0D);
+    Row leafDoubleCol = Row.of(46L, 3L, 1L, 1L, 0.0D, 0.0D);
```

Comment:
@szehon-ho @RussellSpitzer do you know what `leafDoubleCol` is? Is it some kind of metrics that could be changed if the compression codec is changed?

Reply:
Yea I think it include the size, sorry for lack of comment, lets disable zstd in :

```
    Table table =
        catalog.createTable(
            TableIdentifier.of(Namespace.of(database()), tableName()),
            PRIMITIVE_SCHEMA,
            PartitionSpec.unpartitioned(),
            ImmutableMap.of());
```

Reply:
Yeah, I can confirm it's the metric of size. Because of changing to zstd, the sizes are reduced overall. I updated it to the new metrics. We can have a followup PR to change those metric related test with uncompressed parquet.

Reply:
@dbtsai , i looked into it.  I think we have to put it where we write data files, but its a bigger change.  There's a test we use called FileHelpers.writeDataFile() and we need to change the code to take in map of properties.

Then,
```
  public static DataFile writeDataFile(Table table, OutputFile out, List<Record> rows, Map<String, String> properties)
      throws IOException {
    FileFormat format = defaultFormat(table.properties());
    GenericAppenderFactory factory = new GenericAppenderFactory(table.schema());
    properties.forEach(factory::set);

```

but its ok to do in separate pr if you want.

Reply:
Let’s do it in a followup PR to reduce the changes

Reply:
Chat offline with @dbtsai , will make follow up to make metrics take from DataFile.fileSizeInBytes()